beacon-chain/core/blocks/block_operations.go

```diff
@@ -68,6 +69,84 @@ func ProcessEth1DataInBlock(beaconState *pb.BeaconState, block *pb.BeaconBlock)
 	return beaconState
 }
 
+// ProcessBlockHeader validates a block by its header and commits it as latest block.
+//
+// Spec pseudocode definition:
+// def process_block_header(state: BeaconState, block: BeaconBlock) -> None:
+//     # Verify that the slots match
+//     assert block.slot == state.slot
+//     # Verify that the parent matches
+//     assert block.previous_block_root == signing_root(state.latest_block_header)
+//     # Save current block as the new latest block
+//     state.latest_block_header = BeaconBlockHeader(
+//         slot=block.slot,
+//         previous_block_root=block.previous_block_root,
+//         block_body_root=hash_tree_root(block.body),
+//     )
+//     # Verify proposer is not slashed
+//     proposer = state.validator_registry[get_beacon_proposer_index(state)]
+//     assert not proposer.slashed
+//     # Verify proposer signature
+//     assert bls_verify(proposer.pubkey, signing_root(block), block.signature, get_domain(state, DOMAIN_BEACON_PROPOSER))
+func ProcessBlockHeader(
+	beaconState *pb.BeaconState,
+	block *pb.BeaconBlock,
+)(*pb.BeaconState,error){
+	if beaconState.Slot != block.Slot{
+		return nil,fmt.Errorf("state slot: %d is different then block slot: %d",beaconState.Slot,block.Slot)
+	}
+	lbhsr,err:= ssz.SignedRoot(beaconState.LatestBlockHeader )
+	if err!=nil {
+		return nil,err
+	}
+	if !bytes.Equal(block.ParentBlockRoot, lbhsr[:]){
+		return nil,fmt.Errorf("state parentblockroot: %d is different then latest block header signed root: %d",
+		block.ParentBlockRoot,lbhsr)
+	}
+	bBytes,err:= block.Body.Marshal()
+	if err!=nil {
+		return nil,err
+	}
+	bHash,err:=ssz.TreeHash(bBytes)
+	if err!=nil {
+		return nil,err
+	}
+	beaconState.LatestBlockHeader = &pb.BeaconBlockHeader{
+		Slot: block.Slot,
+		PreviousBlockRoot: block.ParentBlockRoot,
+		BlockBodyRoot: bHash[:],
+	}
+	//verify proposer is not slashed
```

Comment:
new line here and format
```suggestion

	// Verify proposer is not slashed
```

Reply:
thanks